server/auth/auth.go

```diff
@@ -626,11 +626,26 @@ func (auth *AuthManager) RequestWithTimeout(user account.AccountID, msg *msgjson
 }
 
 // Penalize signals that a user has broken a rule of community conduct, and that
-// their account should be penalized.
+// their account should be penalized. It logs error and is used by internal
+// callers.
 func (auth *AuthManager) Penalize(user account.AccountID, rule account.Rule) {
+	if err := auth.penalize(user, rule); err != nil {
+		log.Error(err)
+	}
+}
+
+// Ban signals that a user has broken a rule of community conduct, and that
+// their account should be penalized. It returns an error and is used by remote
+// callers.
+func (auth *AuthManager) Ban(user account.AccountID, rule account.Rule) error {
+	return auth.penalize(user, rule)
+}
+
+// penalize signals that a user has broken a rule of community conduct, and that
+// their account should be penalized.
+func (auth *AuthManager) penalize(user account.AccountID, rule account.Rule) error {
```

Comment:
I think there should be a check in this method that rule is one of account.Rule, and something to be penalized for, ie rule > 0 && rule < account.MaxRule.